File: app/src/main/java/ru/dmitrybochkov/rates/view_models/BaseViewModel.java
package ru.dmitrybochkov.rates.view_models;

import androidx.lifecycle.ViewModel;
import io.reactivex.disposables.CompositeDisposable;

/**
 * Created by Dmitry Bochkov on 14.11.2019.
 */
public class BaseViewModel extends ViewModel {

    protected CompositeDisposable compositeDisposable = new CompositeDisposable();

    @Override
    protected void onCleared() {
        super.onCleared();
        compositeDisposable.dispose();
    }
}

File: app/src/main/java/ru/dmitrybochkov/rates/view_models/BaseViewModel.kt
package ru.dmitrybochkov.rates.view_models

import androidx.lifecycle.ViewModel
import io.reactivex.disposables.CompositeDisposable

/**
 * Created by Dmitry Bochkov on 14.11.2019.
 */
open class BaseViewModel : ViewModel() {

    protected var compositeDisposable = CompositeDisposable()

    override fun onCleared() {
        super.onCleared()
        compositeDisposable.dispose()
    }
}